src/services/functionAppService.ts

```diff
@@ -157,51 +158,19 @@ export class FunctionAppService extends BaseService {
 
   public async uploadFunctions(functionApp: Site): Promise<any> {
     Guard.null(functionApp, "functionApp");
-
-    this.log("Deploying serverless functions...");
-
     const functionZipFile = this.getFunctionZipFile();
-
-    if (this.config.provider.deployment.external) {
-      this.log("Updating function app setting to run from external package...");
-      await this.uploadZippedArtifactToBlobStorage(functionZipFile);
-
-      const sasUrl = await this.blobService.generateBlobSasTokenUrl(
-        this.config.provider.deployment.container,
-        this.artifactName
-      );
-
-      const response = await this.updateFunctionAppSetting(
-        functionApp,
-        configConstants.runFromPackageSetting,
-        sasUrl
-      );
-
-      await this.syncTriggers(functionApp, response.properties);
+    // Uploaded to blob storage as artifact for future reference
+    await this.uploadZippedArtifactToBlobStorage(functionZipFile);
+
+    if (this.linuxTarget) {
```

Comment:
This is better - but I think this block should be encapsulated in a separate service so that the `functionAppService` doesn't need to care about the concerns of the OS.

Reply:
Will create a `functionAppPublishService`